api/cloud/oc_cloud.c

```diff
@@ -187,11 +190,19 @@ void
 cloud_update_by_resource(oc_cloud_context_t *ctx,
                          const cloud_conf_update_t *data)
 {
+
+  if (data->ci_server_len == 0) {
+	OC_DBG("[Cloud] got forced deregister via provisioning of empty cis\n");
+	oc_cloud_reset_context(0);
```

Comment:
I think we need to call `oc_cloud_deregister` otherwise the cloud is not informed about deregistration.

Reply:
oc_cloud_reset_context does a deregistration within the method call.

Reply:
Oh. I didn't expect that.